infrastructure/lib/api-ec2-stack.ts

```diff
@@ -116,41 +160,46 @@ export class ApiEc2Stack extends Stack {
                 '*',
             ]
         });
-        const pullPolicy = new Policy(this, 'PullPolicy');
-        pullPolicy.addStatements(
+        const readSecretsPolicyStatment = new PolicyStatement({
+            effect: Effect.ALLOW,
+            actions: [
+                'secretsmanager:GetSecretValue',
+            ],
+            resources: [
+                '*',
+            ]
+        });
+        const littilApiPolicy = new Policy(this, 'PullPolicy');
+        littilApiPolicy.addStatements(
             pullPolicyStatement,
+            readSecretsPolicyStatment,
         );
 
         /**/
         const ec2Instance = new Instance(this, 'ApiInstance', {
             vpc: props.apiVpc,
-            instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.NANO),
+            instanceType: InstanceType.of(InstanceClass.T3A, InstanceSize.MICRO),
             machineImage: new AmazonLinuxImage({
                 generation: AmazonLinuxGeneration.AMAZON_LINUX_2,
-                cpuType: AmazonLinuxCpuType.ARM_64,
+                cpuType: AmazonLinuxCpuType.X86_64,
             }),
             vpcSubnets: {subnetType: SubnetType.PUBLIC},
             securityGroup: ec2SecurityGroup,
-            keyName: keypair.keyName,
+            keyPair: keypair,
             userData,
         });
 
-        pullPolicy.attachToRole(ec2Instance.role);
+        littilApiPolicy.attachToRole(ec2Instance.role);
 
         new CfnEIPAssociation(this, 'Ec2EipAssociation', {
-            eip: elasticIp.ref,
+            eip: props.elasticIp.ref,
             instanceId: ec2Instance.instanceId,
         });
         /**/
 
-        /* Logging. */
-        const apiEc2LoggingStack = new LoggingStack(this, 'ApiEc2LoggingStack', {
-            littil: props.littil,
-            logGroupName: 'BackendApiEc2Logs',
-        });
-
         /* Database access. */
         const databaseSecurityGroup = SecurityGroup.fromSecurityGroupId(this, 'DatabaseSecurityGroup', props.database.securityGroup.id);
-        databaseSecurityGroup.connections.allowFrom(ec2SecurityGroup, Port.tcp(parseInt(props.database.port)));
+        // databaseSecurityGroup.connections.allowFrom(ec2SecurityGroup, Port.tcp(parseInt(props.database.port)));
```

Comment:
do we need to keep this as comment ?

Reply:
Yes and no. It should be as limited as possible, but I don't have time to test it right now. So for now at least I'll add a TODO.